eth/executionclient/execution_client.go

```diff
@@ -306,6 +343,43 @@ func (ec *ExecutionClient) streamLogsToChan(ctx context.Context, logs chan<- Blo
 	}
 }
 
+func (ec *ExecutionClient) streamFinalizedBlocks(ctx context.Context, logs chan<- BlockLogs, fromBlock uint64) (lastBlock uint64, err error) {
+	for {
+		select {
+		case <-ctx.Done():
+			return fromBlock, context.Canceled
+
+		case <-ec.closed:
+			return fromBlock, ErrClosed
+
+		case toBlock, ok := <-ec.blocksChan:
+			if !ok {
+				return lastBlock, fmt.Errorf("blocksChan is closed")
+			}
+			if toBlock < ec.finalizedCheckpointActivationSlot {
```

Comment:
@moshe-blox, @y0sher, 
Not sure do we need this or not.
Is panic suitable here?

Reply:
Lets say we're running a new network, where we sync from block 1 to block 8 , but the fork is only set to block 20, wouldn't this trigger this exit?
also how is block number and slot same in this case?